_includes/manuals/nightly/3.7_migration.md

```diff
@@ -0,0 +1,185 @@
+yum install patch -y
+cd /usr/share/foreman
+
+patch -p1 < /tmp/convert-tasks-patch.diff
+```
+
+The foreman user that has access to your Postgresql database and is used for
+the conversion temporarily needs super user privileges.
+
+```bash
+## postgres@postgres:~  $ su - postgres
+## postgres@postgres:~  $ psql
+psql (9.6.2)
+Type "help" for help.
+
+postgres=# ALTER USER foreman WITH SUPERUSER;
+ALTER ROLE
+postgres=#
+```
+
+#### Migrating
+
+Before proceeding migration, it's a good idea to permanently delete all audit
+records older than few days:
+
+    foreman-rake audits:expire days=10 RAILS_ENV="production"
+
+To permanently delete all reports older than 2 days:
+
+    foreman-rake reports:expire days=2 batch_size=50000 sleep_time=0.000001 RAILS_ENV="production"
+
+The command above deletes records in batches of 50k items, it can take a lot of
+time to process all records. In both cases, progress can be seen in the
+`production.log`.
+
+To migrate the data from one database to the other, we will use a rake task.
+You might already be familiar to this task from upgrading Foreman to a newer
+version.
+
+Command `foreman-rake -T` lists all available tasks. We will be using
+`foreman-rake db:convert:prod2dev` to migrate the data.
+
+Foreman internally uses a development framework called Rails. In Rails, you can
+have a specific configuration for e.g. a development and a production
+environment. Our prod2dev rake task assumes, that the source database is
+configured as the production database and the new target database is configured
+as the development database. All data in the development database will be
+overwritten.
+
+You can configure the new database as Foreman's new development database by
+adding a new connection in `/etc/foreman/database.yml`.
+
+It may look like this:
+
+```yaml
+# The new target database
+development:
+  adapter: postgresql
+  host: postgres.example.com
+  port: 5432
+  database: foreman
+  username: foreman
+  password: "foreman"
+  pool: 25
+
+# The old source database
+production:
+  adapter: mysql2
+  host: mysql.example.com
+  port: 3306
+  database: foreman
+  username: foreman
+  password: "foreman"
+  pool: 25
+```
+
+The new database needs to be set up with all the tables required for Foreman to
+operate, so in order to prepare the new database to receive the actual data, we
+need to run all of Foreman's database migrations.
+
+```sh
+forman-rake db:migrate RAILS_ENV=development
+```
+
+Next, we can finally migrate all the data. This will take a long time depending
+on the size of your database. It is highly recommended to run this in a screen
+or tmux session.
+
+```sh
+forman-rake db:convert:prod2dev
+
+```
+
+#### Wrapping up
+
+Postgresql uses sequences in order to store the next free primary key id for a
+table. These sequences do not get updated by the rake tasks and have to be
+adjusted manually. On MySQL, this is not needed as the concept of sequences is
```

Comment:
Should we add this script to the migration script in case the target is pg?

Reply:
I think this would be great to have. Or another rake task that runs the commands on the database if it's mysql. Updating the sequences should be rather safe to do.

Reply:
Into which migration script, `prod2dev` is not supposed to do MySQL migrations.